scripts/mlrun.sh

```diff
@@ -17,15 +24,35 @@ kubectl create namespace mlrun
 helm repo add mlrun-ce https://mlrun.github.io/ce
 helm repo update
 
-# Create docker registry secrets:
-kubectl --namespace mlrun create secret docker-registry registry-credentials \
-    --docker-server $DOCKER_SERVER \
-    --docker-username $DOCKER_USERNAME \
-    --docker-password=$DOCKER_PASSWORD
+# Set up a local docker registry on the minikube cluster only if docker registry is not provided:
+if [ $LOCAL_DOCKER_REGISTRY_FLAG -eq 1 ]; then
+  echo "No docker registry URL provided, setting up a local docker registry on the minikube cluster..."
+  minikube addons enable registry
+  DOCKER_REGISTRY_URL="http://$(minikube ip):5000"
+  DOCKER_SERVER="$(minikube ip):5000"
+
+  # wait for the registry to be ready:
+  while ! kubectl get pods -n kube-system | grep -q 'registry-'; do
+    echo "Waiting for local docker registry to be ready..."
+    sleep 5
+  done
+fi
+
+# Create a configmap for local docker registry:
+kubectl create configmap registry-config \
+  --namespace=$MLRUN_NAMESPACE \
+  --from-literal=insecure_pull_registry_mode=enabled \
+  --from-literal=insecure_push_registry_mode=enabled
+
+# Create docker registry secrets: FAILED
+kubectl --namespace $MLRUN_NAMESPACE create secret docker-registry registry-credentials \
+    --docker-server "$DOCKER_SERVER" \
+    --docker-username "$DOCKER_USERNAME" \
+    --docker-password="$DOCKER_PASSWORD"
 
 # Install mlrun ce:
-helm --namespace mlrun install mlrun-ce --wait --timeout 960s \
-  --set global.registry.url=$DOCKER_REGISTRY_URL \
+helm --namespace $MLRUN_NAMESPACE install mlrun-ce --wait --timeout 960s \
+  --set global.registry.url="$DOCKER_REGISTRY_URL" \
```

Comment:
how did it work before without the ""?

Reply:
It worked; it's just for safety measures.